rules/library.bzl

```diff
@@ -536,6 +540,62 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
                 opts = repr(v),
             ))
 
+    # Generate code for `.intentdefinition` source files and forward the original plist as data
+    for intent in intent_sources:
+        intent_name = "%s_%s_gen" % (name, intent)
+        testonly = kwargs.get("testonly", False)
+
+        # Avoid swift_intent_library and objc_intent_library because they wrap the generated code in a new module
+        #
+        # If other Swift sources are present, generate Swift intent code, otherwise use Obj-C.
+        # This mimics the behavior for INTENTS_CODEGEN_LANGUAGE="automatic" in Xcode.
```

Comment:
Nice I think this follows the general design around `rules_ios` to mimic Xcode behavior

Reply:
+1 I like how you integrated it here